runtime/arc.js

```diff
@@ -159,7 +169,30 @@ class Arc {
     return s;
   }
 
-  connectParticleToView(particleHandle, particle, name, targetView) {
+  instantiate(recipe) {
+    assert(recipe.isResolved(), 'Cannot instantiate an unresolved recipe');
+    let {views, particles, slots} = recipe.mergeInfo(this._activeRecipe);
+    let recipeViewMap = new Map();
+    views.forEach(recipeView => {
+      let view;
+      if (recipeView.create) {
```

Comment:
as discussed offline, i switched to creating the new ones, updating their ids and create flag, and asserting all the rest already exist.